src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -531,18 +528,20 @@ def post_process_panoptic_segmentation(
                 # this is a map between stuff and segments id, the used it to keep track of the instances of one class
                 for k in range(pred_labels.shape[0]):
                     pred_class = pred_labels[k].item()
-                    # check if pred_class is not a "thing", so it can be merged with other instance. For example, class "sky" cannot have more then one instance
-                    is_stuff = not is_thing_map[pred_class]
+                    # check if pred_class should be fused. For example, class "sky" cannot have more then one instance
+                    should_fuse = False
+                    if label_ids_to_fuse is not None:
+                        should_fuse = pred_class in label_ids_to_fuse
```

Comment:
Forcing `label_ids_to_fuse` early on to be a `set` (maybe empty) is IMO better, since you can start relying on the variable type instead of having `if ... None` everywhere.

```python
  if label_ids_to_fuse is None:
            logger.warning("`label_ids_to_fuse` unset. No instance will be fused.")
            label_ids_to_fuse = set()
```
And then here.:
```python
should_fuse = pred_class in label_ids_to_fuse
``` 
No check necessary, you already changed `label_ids_to_fuse` to be a set.

Again the only reason why you don't use `my_function(....label_ids_to_fuse=set())` is because `set` is mutable.
With immutable types (like `tuple`) it would be preferable to use a default argument imo.

> Note: Technically you can use `frozenset` as the default argument but my personal take is that keeping things simple with `None` as default argument is easy enough and understandable too `frozenset` is not that widely used. 
To be fair, `set` itself is probably not used that much either, but it does have a generator comprehension at least `myset = {letter for letter in "hello"}`)